Feature/partners listing applications

dominikx96 opened the pull request:
Resolves: #730, #404, #392

**Next steps (backend not ready yet, #811  ):**
#404:
- Replace hardcoded data:
    - Application Status
    - Application Language
    - Application Submission Type

dominikx96 commented:
@slowbot 

Changes are implemented, also filtering in the applications table.
We still wait for backend fields.

dominikx96 commented:
@slowbot header has been updated (added first, last name & confirmation ID), but I used the current text style in the header component, is it ok for you?

jaredcwhite commented:
Hey @dominikx96, we have the ViewItem component in both `blocks` and `elements` now, so we should pick one category and make sure it's up to date.

jaredcwhite commented:
Looking more closely, the one in blocks is the one being exported out of ui-components, so I guess we could just remove the one in elements?

dominikx96 commented:
@jaredcwhite yes, I removed the duplicated component, thanks for noticing me!

@slowbot before merge I have few questions about newly added fields:

- When the user does not check "Send my mail to a different address" in the application form: should we show the mailing address section with the same data as in "RESIDENCE ADDRESS" or show something like "Same as residence address"?

- A "Middle Name" field - should it be visible if is empty?

- We added fields for application language, submission type, application status, but what about old applications (existed before this change)?
  - LANGUAGE: should we hide this cell or show English as default (correct me if I'm wrong, but before all applications were in EN ?)
  - SUBMISSION TYPE: should be hidden?
  - STATUS: I think it should be "submitted" as the default
